plugins/org.komodo.shell-api/src/org/komodo/shell/api/KomodoObjectLabelProvider.java

```diff
@@ -148,5 +151,19 @@
      *        the workspace status (never <code>null</code>)
      */
     void setWorkspaceStatus( final WorkspaceStatus status );
+    
+	 /**
+     * @return Returns a node names which should be omitted from display path
+     *         (Never <code>null</code>. Returns empty list if not grouping nodes are available)
+     */
+	List<String> skippedPathSegmentNames();
+	
+    /**
+     * Get the type display string for a KomodoObject
+     * @param uow the transaction (never <code>null</code>)
+     * @param kObj the KomodoObject (never <code>null</code>)
+     * @return the type display string (never returns <code>null</code>)
```

Comment:
I think the return value **must** allow `null` to signify that the label provider does not know how to come up with a type display for the specified object. See `WorkspaceStatusImpl.setLabelProvider`.